exec/java-exec/src/main/java/org/apache/drill/exec/compile/ClassTransformer.java

```diff
@@ -264,7 +265,13 @@ public boolean equals(Object obj) {
         final ClassNames nextPrecompiled = nextSet.precompiled;
         final byte[] precompiledBytes = byteCodeLoader.getClassByteCodeFromPath(nextPrecompiled.clazz);
         final ClassNames nextGenerated = nextSet.generated;
-        final ClassNode generatedNode = classesToMerge.get(nextGenerated.slash);
+        Pair<byte[], ClassNode> classNodePair = classesToMerge.remove(nextGenerated.slash);
+        final ClassNode generatedNode;
+        if (classNodePair != null) {
+          generatedNode = classNodePair.getValue();
+        } else {
+          generatedNode = null;
```

Comment:
What does it mean by "generatedNode = null"? Will it cause problem in subsequent call at line 288?